java/core/src/test/java/sleeper/core/record/process/status/ProcessRunsTest.java

```diff
@@ -190,4 +194,65 @@ public void shouldIgnoreExtraFinishedStatus() {
         assertThat(runs.isFinished()).isTrue();
     }
 
+    @Test
+    void shouldNotCreateProcessRunIfStatusUpdateNotFlaggedAsStartOfRun() {
+        // Given
+        ProcessStatusUpdate notStartedUpdate = () -> Instant.parse("2022-09-24T09:23:30.001Z");
+
+        // When
+        ProcessRuns runs = runsFromUpdates(notStartedUpdate);
+
+        // Then
+        assertThat(runs.getRunList()).isEmpty();
+    }
+
+    @Test
+    void shouldCreateProcessRunFromTwoStartedUpdatesWhenOneIsFlaggedAsStartOfRun() {
+        // Given
+        ProcessStartedStatusWithStartOfRunFlag startedStatusNotStartOfRun = updateAndStartTimeNotStartOfRun(
+                Instant.parse("2022-09-24T08:23:30Z"), Instant.parse("2022-09-24T08:23:30.001Z"));
+        ProcessStartedStatus startedStatus = updateAndStartTime(
+                Instant.parse("2022-09-24T09:23:30Z"), Instant.parse("2022-09-24T09:23:30.001Z"));
+
+        // When
+        ProcessRuns runs = runsFromUpdates(startedStatusNotStartOfRun, startedStatus);
+
+        // Then
+        assertThat(runs.getRunList())
+                .containsExactly(ProcessRun.started(DEFAULT_TASK_ID, startedStatus));
+    }
+
+    @Test
+    void shouldCreateProcessRunWithCustomStatusUpdatePartOfRun() {
+        // Given
+        ProcessStartedStatus startedStatus = updateAndStartTime(
+                Instant.parse("2022-09-24T09:23:30Z"), Instant.parse("2022-09-24T09:23:30.001Z"));
+        CustomProcessStatus customStatus = partOfRunWithUpdateTime(Instant.parse("2022-09-24T10:23:30Z"));
+
+        // When
+        ProcessRuns runs = runsFromUpdates(startedStatus, customStatus);
+
+        // Then
+        assertThat(runs.getRunList())
+                .containsExactly(ProcessRun.builder()
+                        .taskId(DEFAULT_TASK_ID)
+                        .startedStatus(startedStatus)
+                        .statusUpdate(customStatus)
+                        .build());
```

Comment:
Can we test that we can retrieve the custom status update from the run by its class?

We could do that in a nested test class that defines the behaviour for retrieving a status update from a run by its class, eg. `run.getLastStatusOfType(CustomProcessStatus.class)`.

We could also test the case where there are multiple updates of the same class in the same run.